pkg/resources/kafka/configmap.go

```diff
@@ -33,74 +29,105 @@ import (
+			if err := config.Set("cruise.control.metrics.reporter.security.protocol", "SSL", ""); err != nil {
+				log.Error(err, "setting cruise.control.metrics.reporter.security.protocol in broker configuration resulted an error")
+			}
+			if err := config.Set("cruise.control.metrics.reporter.ssl.truststore.location", clientKeystorePath+"/"+v1alpha1.TLSJKSTrustStore, ""); err != nil {
+				log.Error(err, "setting cruise.control.metrics.reporter.ssl.truststore.location in broker configuration resulted an error")
+			}
+			if err := config.Set("cruise.control.metrics.reporter.ssl.keystore.location", clientKeystorePath+"/"+v1alpha1.TLSJKSKeyStore, ""); err != nil {
+				log.Error(err, "setting cruise.control.metrics.reporter.ssl.keystore.location parameter in broker configuration resulted an error")
+			}
+			if err := config.Set("cruise.control.metrics.reporter.ssl.keystore.password", clientPass, ""); err != nil {
+				log.Error(err, "setting cruise.control.metrics.reporter.ssl.keystore.password parameter in broker configuration resulted an error")
+			}
+		}
+	}
+
+	// Add Cruise Control Metrics Reporter configuration
+	if err := config.Set("metric.reporters", "com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter", ""); err != nil {
+		log.Error(err, "setting metric.reporters in broker configuration resulted an error")
+	}
+	bootstrapServers := getInternalListener(r.KafkaCluster.Spec.ListenersConfig.InternalListeners, id, r.KafkaCluster.Spec.GetKubernetesClusterDomain(), r.KafkaCluster.Namespace, r.KafkaCluster.Name, r.KafkaCluster.Spec.HeadlessServiceEnabled)
+	if err := config.Set("cruise.control.metrics.reporter.bootstrap.servers", bootstrapServers, ""); err != nil {
+		log.Error(err, "setting cruise.control.metrics.reporter.bootstrap.servers in broker configuration resulted an error")
+	}
+	if err := config.Set("cruise.control.metrics.reporter.kubernetes.mode", true, ""); err != nil {
+		log.Error(err, "setting cruise.control.metrics.reporter.kubernetes.mode in broker configuration resulted an error")
+	}
+
+	// Kafka Broker configuration
+	if err := config.Set("broker.id", id, ""); err != nil {
+		log.Error(err, "setting broker.id in broker configuration resulted an error")
+	}
+
+	// Storage configuration
+	storageConf := generateStorageConfig(bConfig.StorageConfigs)
+	if storageConf != "" {
+		if err := config.Set("log.dirs", storageConf, ""); err != nil {
+			log.Error(err, "setting log.dirs in broker configuration resulted an error")
+		}
+	}
+
+	// Add listener configuration
+	advertisedListenerConf := generateAdvertisedListenerConfig(id, r.KafkaCluster.Spec.ListenersConfig, extListenerStatuses, intListenerStatuses, controllerIntListenerStatuses)
+	if len(advertisedListenerConf) > 0 {
+		if err := config.Set("advertised.listeners", advertisedListenerConf, ""); err != nil {
+			log.Error(err, "setting advertised.listeners in broker configuration resulted an error")
+		}
+	}
```

Comment:
Nit: I would move this up just after line 40. In that case the listener specific configs can be together:)

Reply:
Agreed. Will fix. Ty.